File: JBGE_RealityKit/JBGE_RealityKit/GameEngine/Core/Enumerations.swift
//
//  Enumerations.swift
//  JBGE_RealityKit
//
//  Created by Tomohiro Kadono on 2026/01/12.
//

// Image direction to search in the atlas for creating animations that are made up of multiple sprites
public enum SheetOrientation { case Horizontal, Vertical }
// General horizontal alignment definitions
public enum HorizontalAlignment { case Left, Center, Right }
// General vertical alignment definitions
public enum VerticalAlignment { case Top, Middle, Bottom }
// General orientation definitions
public enum Orientation { case Horizontal, Vertical }
// General speed identifiers
public enum MovementSpeed { case VerySlow, Slow, Normal, Fast, VeryFast }
// 8 Directions
public enum Direction { case South, North, East, West, NorthEast, SouthEast, SouthWest, NorthWest }
// Gamepad component definitions
public enum GamePad {
    case SDPadNorth, DPadSouth, DPadWest, DPadEast, DPadNorthEast, DPadSouthEast, DPadSouthWest, DPadNorthWest, ButtonNorth, ButtonSouth, ButtonWest, ButtonEast, ButtonLeftShoulder, ButtonRightShoulder, ButtonLeftTrigger, ButtonRightTrigger, ButtonLeftStick, ButtonRightStick, ButtonStart, ButtonSelect
}

File: JBGE_Unity/Assets/JBGE_Unity/GameEngine/Enumerations.cs
using System;

namespace JBGE {
	/*
	public enum GameResolutionType {
		RES_1920x1080xScale1, RES_1920x1080xScale5, RES_1280x800xScale1, RES_1280x800xScale5
	}*/
	// Image direction to search in the atlas for creating animations that are made up of multiple sprites
	public enum SheetOrientation { Horizontal, Vertical }
	// General horizontal alignment definitions
	public enum HorizontalAlignment { Left, Center, Right }
	// General vertical alignment definitions
	public enum VerticalAlignment { Top, Middle, Bottom }
	// General orientation definitions
	public enum Orientation { Horizontal, Vertical }
	// General speed identifiers
	public enum MovementSpeed { VerySlow, Slow, Normal, Fast, VeryFast }
	public enum Direction { South, North, East, West, NorthEast, SouthEast, SouthWest, NorthWest }
	public enum GamePad {
		DPadNorth, DPadSouth, DPadWest, DPadEast, DPadNorthEast, DPadSouthEast, DPadSouthWest, DPadNorthWest, ButtonNorth, ButtonSouth, ButtonWest, ButtonEast, ButtonLeftShoulder, ButtonRightShoulder, ButtonLeftTrigger, ButtonRightTrigger, ButtonLeftStick, ButtonRightStick, ButtonStart, ButtonSelect
	}
}
